### PR TITLE
mig-controller : PR Testing

### LINKED ISSUE #173: PR Testing
_[Tracking issue for PR testing pipeline]_

Enable automatic PR building with Github webhook.

Repos to build : 
- mig-controller
- mig-operator
- mig-ui  [Low priority]

### PR DESCRIPTION
**Pull request by pranavgaikwad**
Fixes #173 

Adds : 
- New pipeline specific to PR builds

### CODE REVIEW

File: pipeline/common_stages.groovy [hunk 1]
```diff
@@ -390,7 +390,9 @@ def deploy_mig_controller_on_both(
     stage('Build mig-controller image and deploy on both clusters') {
       steps_finished << 'Build mig-controller image and deploy on both clusters'
       // Create custom mig-controller docker image if building a different mig-controller repo/branch
-      if ("${MIG_CONTROLLER_REPO}" != "https://github.com/konveyor/mig-controller.git") {
+      if ("${MIG_CONTROLLER_REPO}" != "https://github.com/konveyor/mig-controller.git" || 
+          "${MIG_CONTROLLER_BRANCH}" != "master"
+        ) {
```
Comment (pranavgaikwad) [inline pipeline/common_stages.groovy:395]: If we want to build a controller image with konveyor/mig-controller repo but with different branch than master  `konveyor/mig-controller:release-1.1`

File: pipeline/common_stages.groovy [hunk 2]
```diff
@@ -453,7 +457,7 @@ def deploy_mig_controller_on_both(
         ansiColor('xterm') {
           ansiblePlaybook(
             playbook: 'mig_controller_deploy.yml',
-            extras: "-e mig_controller_host_cluster=${mig_controller_dst} -e mig_controller_ui=${MIG_CONTROLLER_UI}",
+            extras: "-e mig_controller_image=${mig_controller_img} -e mig_controller_version=${mig_controller_tag} -e mig_controller_host_cluster=${mig_controller_dst} -e mig_controller_ui=${MIG_CONTROLLER_UI}",
```
Comment (pranavgaikwad) [inline pipeline/common_stages.groovy:460]: allow overriding mig-controller image in MigrationController configuration - custom PRs

File: pipeline/common_stages.groovy [hunk 3]
```diff
@@ -463,8 +467,47 @@ def deploy_mig_controller_on_both(
   }
 }
 
+/*
+  Builds operator images, creates Quay
+  application, pushes application
 
-def execute_migration(e2e_tests, source_kubeconfig, target_kubeconfig) {
+  directory [string] : Directory of operator repo to build
+*/
```
Comment (pranavgaikwad) [inline pipeline/common_stages.groovy:475]: build custom operator image in following function. 

use env variables `MIG_OPERATOR_REPO` and `MIG_OPERATOR_BRANCH` 

each build increments the Quay application version number by 1

File: pipeline/utils.groovy [hunk 4]
```diff
@@ -143,6 +143,81 @@ def copy_public_keys() {
   }
 }
 
+/*
```
Comment (pranavgaikwad) [inline pipeline/utils.groovy:146]: parse comment message to find mentions for operator / controller PR number

set environment variables based on the comment

File: pipeline/mig-controller-pr-builder.groovy [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+booleanParam(defaultValue: false, description: 'Deploy using downstream images', name: 'USE_DOWNSTREAM'),
+booleanParam(defaultValue: false, description: 'Enable debugging', name: 'DEBUG'),
+booleanParam(defaultValue: true, description: 'Clean up workspace after build', name: 'CLEAN_WORKSPACE')])])
+
+
+// true/false build parameter that defines if we use OLM to deploy mig operator on OCP4
+USE_OLM = params.USE_OLM
+USE_DISCONNECTED = false
+E2E_DEPLOY_ONLY = params.E2E_DEPLOY_ONLY
+MIG_CONTROLLER_UI = false
```
Comment (fbladilo) [inline pipeline/mig-controller-pr-builder.groovy:42]: I think this one is redundant, already defined as a pipeline PARAM above

**Comment by fbladilo** [comment]
LGTM , tested with all current prod pipelines.

**Pull request merged**